src/Voter.jsx

```diff
@@ -0,0 +1,38 @@
+import { addVote } from './Api'
+import { useEffect, useState } from "react"
+import { Link } from 'react-router-dom';
+
+
+function Voter({article_id, currentVotes, setVotes}) {
+const [hasVoted, setHasVoted] = useState(false);
+
+
+const updateVote = () => {
+let vote = 1    
+if (hasVoted) {vote = 0}
+setVotes(vote)
+addVote(article_id, vote? 1 : -1)
```

Comment:
With regards to error handling on this one, we're only interested in if this request fails and goes to the _catch_. It's just about demonstrating to a user that it's failed, this can be done by reverting the vote count back to undo the optimistic rendering - and you could add in a little `<p>` tag to say something